shell.nix

```diff
@@ -1,73 +1,78 @@
-{ packageSet ? import ./default.nix { rev = "in-nix-shell"; }
+{ crossSystem ? null
+, system ? builtins.currentSystem
+, config ? { allowUnfreePredicate = (import ./lib.nix).unfreePredicate; }
+, rev ? "in-nix-shell"
+, sourcesOverride ? { }
+, pkgs ? import ./nix { inherit crossSystem config sourcesOverride; }
+, _pkgs ? import ./nix/pkgs { inherit rev pkgs; checkMaterialization = false; }
 }:
-with packageSet;
 let
+  inherit (pkgs) python3 sphinxcontrib-haddock nix-pre-commit-hooks lib utillinux nixpkgs-fmt;
+  inherit (pkgs.stdenv) isDarwin isLinux;
+  inherit (pkgs.lib) optionals optionalString;
```

Comment:
I think I would mildly prefer to use some things qualified rather than dumping everything into the scope. Some of that is just convention, e.g. I see `lib.whatever` and I know where it's from.

Reply:
I think you don't like qualified access ;)

Reply:
fair enough, might have overdone it there :sweat_smile: i dialed it down now..